container/image.bzl

```diff
@@ -456,3 +462,10 @@ def container_image(**kwargs):
   if "entrypoint" in kwargs:
     kwargs["entrypoint"] = _validate_command("entrypoint", kwargs["entrypoint"])
   container_image_(**kwargs)
+  # Also create a "name.top_layer" container_layer target
+  top_layer_attrs = {}
+  for key in _layer.attrs.keys():
+    if key in kwargs:
+      top_layer_attrs[key] = kwargs[key]
+  top_layer_attrs["name"] = kwargs["name"] + ".top_layer"
+  container_layer_(**top_layer_attrs)
```

Comment:
I was thinking more that `container_image` would *consume* the `.top_layer` itself here, otherwise we're effectively creating the top layer two different ways, which isn't what I'd expect as a user.

Reply:
Actually `container_image` is already doing so in the implementation (use container_layer's implementation for generating the top_layer). 

The only thing I am worry about is about the other user of `container_image`'s **implementation** (not the rule itself). Since we move the logic for generating the `top_layer` into the macro:
- people use the implementation will NOT not get the `top_layer` built
- at the same time people cannot reuse `container_image` macro in their implementations. 

WDYT?

Reply:
Sorry for the delay.  I think it is fine (and actually should be by-design) that `lang_image` doesn't expose `.top_layer`.  For `lang_image`, it is my expectation that the layering choice is an implementation detail that could change at any time.

Reply:
Hi Matt, 

So sorry for not working lately. My main concern about splitting the logic `container_image` into pieces in macro is, internally we extend the implementation of `container_image`, and if we split part of its logic in macro, we can no longer re-use it easily by extending the implementation. 

Given that we probably need to migrate to the [new package manager](https://github.com/GoogleCloudPlatform/base-images-docker/tree/master/package_managers), I don't think this PR is urgent for us. So I am also OK to close this PR and park the changes for now.